tools/iree-check-module-main.cc

```diff
@@ -147,7 +147,7 @@ iree_status_t Run(iree_allocator_t host_allocator, int* out_exit_code) {
 
 }  // namespace
 
-extern "C" int main(int argc, char** argv) {
+int main(int argc, char** argv) {
```

Comment:
This is probably fine for these tools, but it makes `iree::main` a symbol with external linkage. One potential problem is this if tool will load a shared lib which is also defines `iree::main` as external symbol and interposition will kick in. Probably should add static/anon namespace.

Reply:
Good point... feel free to make a follow-up giving `iree::main` internal linkage. I am not worried about name clashes since we "own" the iree namespace, but it is a good idea to give internal functions internal linkage, as it results in slightly more optimized code.